pylint/checkers/base.py

```diff
@@ -2098,6 +2162,52 @@ def _name_became_keyword_in_version(name, rules):
                 return ".".join(str(v) for v in version)
         return None
 
+    @staticmethod
+    def _assigns_typevar(node: Optional[nodes.NodeNG]) -> bool:
+        """Check if a node is assigning a TypeVar"""
+        if isinstance(node, astroid.Call):
+            inferred = utils.safe_infer(node.func)
+            if (
+                inferred
+                and isinstance(inferred, astroid.ClassDef)
+                and inferred.qname() == TYPING_TYPE_VAR_QNAME
+            ):
+                return True
+        return False
+
+    def _check_typevar_variance(
+        self, name: str, node: nodes.AssignName
+    ) -> Optional[Tuple[str, str, str]]:
+        """Check if a TypeVar has a variance, and if so if it is included in its name.
+        Returns the args for the message to be displayed"""
+        _co, _contra = "_co", "_contra"
+        if self._typevar_naming_style == "UPPER_CASE":
+            _co, _contra = "_CO", "_CONTRA"
+
+        if isinstance(node.parent, nodes.Assign):
+            keywords = node.assign_type().value.keywords
+        elif isinstance(node.parent, nodes.Tuple):
+            keywords = (
+                node.assign_type().value.elts[node.parent.elts.index(node)].keywords
+            )
+
```

Comment:
```suggestion
            )
        else:
            return None
```
Seems like we don't handle the case where it's not a Assign not a Tuple. But I guess it never happens ?